app/controllers/orangelight/browsables_controller.rb

```diff
@@ -1,59 +1,179 @@
 # frozen_string_literal: true
 
 class Orangelight::BrowsablesController < ApplicationController
-  # GET /orangelight/names
-  # GET /orangelight/names.json
-  def index
+  def update_facet
+    if browsing_names?
+      @facet = 'author_s'
+    elsif browsing_subjects?
+      @facet = 'subject_facet'
+    elsif browsing_titles?
+      @facet = 'name_title_browse_s'
+    end
+  end
+
+  # All of this needs to be restructured into a QueryService Class
```

Comment:
I still stand by this, but I can split this work into a pull request if that is acceptable.